src/BaseView_TEST.cc

```diff
@@ -152,30 +155,27 @@ TEST_F(BaseViewTest, AddEntities)
       modelNameView.NewEntities().end());
 
   auto e1ConstData = modelNameView.EntityComponentConstData(e1);
-  EXPECT_EQ(e1, std::get<Entity>(e1ConstData));
-  EXPECT_EQ(&e1ModelComp, std::get<const components::Model *>(e1ConstData));
-  EXPECT_EQ(&e1NameComp, std::get<const components::Name *>(e1ConstData));
+  ASSERT_EQ(2u, e1ConstData.size());
+  EXPECT_EQ(&e1ModelComp, e1ConstData[0]);
+  EXPECT_EQ(&e1NameComp, e1ConstData[1]);
 
   auto e1Data = modelNameView.EntityComponentData(e1);
-  EXPECT_EQ(e1, std::get<Entity>(e1Data));
-  EXPECT_EQ(&e1ModelComp, std::get<components::Model *>(e1Data));
-  EXPECT_EQ(&e1NameComp, std::get<components::Name *>(e1Data));
+  EXPECT_EQ(&e1ModelComp, e1Data[0]);
+  EXPECT_EQ(&e1NameComp, e1Data[1]);
 
-  auto e2ConstData = modelNameView.EntityComponentConstData(e2);
-  EXPECT_EQ(e2, std::get<Entity>(e2ConstData));
-  EXPECT_EQ(&e2ModelComp, std::get<const components::Model *>(e2ConstData));
-  EXPECT_EQ(&e2NameComp, std::get<const components::Name *>(e2ConstData));
+  auto e2ConstData = modelNameView .EntityComponentConstData(e2);
+  EXPECT_EQ(&e2ModelComp, e2ConstData[0]);
+  EXPECT_EQ(&e2NameComp, e2ConstData[1]);
 
   auto e2Data = modelNameView.EntityComponentData(e2);
```

Comment:
nit: make sure the vector has 2 elements in it before indexing it to avoid out of bounds access segfault
```suggestion
  auto e2Data = modelNameView.EntityComponentData(e2);
  ASSERT_EQ(2u, e2Data.size());
```